### PR TITLE
[#11182] Proof-of-submission file: Add a warning if no answers were saved

### LINKED ISSUE #11182: Proof-of-submission file: Add a warning if no answers were saved
While we figure out what's going on in #10994 (there may be multiple causes), in the interim, I would like to add a clear warning in the proof-of-concept file if there are no answers in it.

image

e.g., `WARNING! No answers were saved. Please try again.`

Reason: Without such a clear message, the user might think everything is fine because s/he doesn't realize the saved answers should be there in the file as well.

Even better if we give that warning as a dialog box i.e., not give a proof-of-concept file at all.

### PR DESCRIPTION
**Pull request by ypinhsuan**
Fixes #11182

**Outline of Solution**
It seems like the reason for empty proof is because all the responses were not saved. 

Updated:
Students will be unable to download the proof of submission if all questions are not saved.
image

If partial saving is done (ie some questions are saved), students can download the proof of submission and an error message will be displayed for questions unsaved.
image
image

@damithc do you want to allow the students to download the proof if there is a question saved properly? (ie some questions are saved and some are not, so the proof will not be entirely empty)
I think an alternative is to display in the proof of submission which questions are not saved.

### CODE REVIEW

**Comment by damithc** [comment]
Thanks for taking this issue @ypinhsuan 
If the saving is not complete (even for partial success), we can show the dialog as per your screenshot, but also allow downloading the file. The file too can contain the error information. So, the error is indicated in both forms -- less chance for the user to miss it.
Is that possible?

**Comment by ypinhsuan** [comment]
Yep that sounds good! I will work on displaying error in both forms.

**Comment by ypinhsuan** [comment]
Update: 
Student can download the proof if at least one question is saved successfully. (There are a total of 5 qns in this screenshot)
image

Error message will be displayed in the proof for unsaved questions
image

**Comment by damithc** [comment]
Looks good @ypinhsuan Some suggestions:
1. `some error` will be replaced by a more meaningful error message?
2. In the dialog, can rephrase as 
   `ERROR! The response submissions to the following questions were unsuccessful.`
2. In the file, can phrase as `ERROR! The response submission to this question was unsuccessful.`
3. The dialog title: `Some response submissions failed!`

**Comment by ypinhsuan** [comment]
For point 1, the "some error" message was created by me to simulate a situation where saving has failed, the users will not be seeing this. In the actual case where saving failed, this message will be replaced by an actual error message telling the user the cause of failure.

**Comment by ypinhsuan** [comment]
Udpated the message displayed to users in dialog box and proof of submission
image
image

**Comment by damithc** [comment]
> Udpated the message displayed to users in dialog box and proof of submission

Good. In fact, can replace `was unsuccessful` with `failed` in each place.

**Comment by ypinhsuan** [comment]
I have changed `was unsuccessful` to `failed` and updated the UI screenshots in the PR description.

File: src/web/app/pages-session/session-submission-page/session-submission-page.component.ts [hunk 1]
```diff
@@ -90,6 +90,7 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
   formattedSessionClosingTime: string = '';
```
Comment (rrtheonlyone) [inline src/web/app/pages-session/session-submission-page/session-submission-page.component.ts:90]: You do not need to change this file at all. The number of questions can be obtained from the length of `questions: QuestionSubmissionFormModel[]`

**Review by rrtheonlyone — APPROVED** [review]
Nice one! Good to go, asking prof @damithc for final UI review

**Comment by damithc** [comment]
> Good to go after the below change, asking prof @damithc for final UI review

Looks good to me 👍

**Pull request merged**